### PR TITLE
Map visualization

### LINKED ISSUE #61: [Map] Integrate standalone Map Visualization
For epic - [Map Visualization](https://app.zenhub.com/workspaces/digital-public-goods---technical-roadmap---5f9c4ed1abaaf60010f1bf2e/issues/unicef/publicgoods-roadmap/34) 
As [MVP](https://app.zenhub.com/workspaces/digital-public-goods---technical-roadmap---5f9c4ed1abaaf60010f1bf2e/issues/unicef/publicgoods-roadmap/48) is ready the next step is to implement standalone visualization into [digitalpublicgoods.net/](https://digitalpublicgoods.net/)**map** 🚀 

- [x] Copy [repo](https://github.com/Rolikasi/digital-public-goods-map) into [publicgoods-scipts/packages](https://github.com/unicef/publicgoods-scripts/tree/main/packages)
 - [x] update moveFiles.bash
 - [x] update github actions workflow 
 
[Current Implementation](https://digital-public-goods-map.vercel.app/)

### LINKED ISSUE #65: [Map] add navigation header and footer
For epic - [Map Visualization](https://app.zenhub.com/workspaces/digital-public-goods---technical-roadmap---5f9c4ed1abaaf60010f1bf2e/issues/unicef/publicgoods-roadmap/34)

The map should be more tightly integrated with the DPGA website by embedding it in a page that has the header and footer matching the rest of the website.

**Note**
For the desktop version I suggest to reduce the size of the footer and keep it in the info container on the left side of the page.
footer

### PR DESCRIPTION
**Pull request by Rolikasi**
**Changes**
I added [repo](https://github.com/Rolikasi/digital-public-goods-map) for map visualization as a package
made changes to  `.github/workflows/auto.yml` and  `scripts/moveFiles.bash`
Before merging we need to add secrets for `NEXT_PUBLIC_MAPBOX_TOKEN` and `NEXT_PUBLIC_SHEET_ID` in `Actions secrets` in github

**Workflow**
Map package build nextjs app with static jsons for story and dpgs data, then exports static version of map in publicgoods-website/map folder. 

**Result**
Map is correctly displayed under `localhost:8080/map/` (digitalpublicgoods.net/map)
publicgoods-map

**Future**
We need to create triggers to rerun script below after changes are made in [gdoc](https://docs.google.com/spreadsheets/d/1t75gYVhdUjPD1532DbPYN49FLXFhpRwEBFiS4Hbk6_Q/edit#gid=728344896)
```
      - name: Build packages/map
        run: cd packages/map && npm install && npm run build
      - name: Run moveFiles.bash
        run: bash scripts/moveFiles.bash
```

Closes #61 
Closes #65
Kindly review changes @nathanfletcher @lacabra

### CODE REVIEW

File: packages/map/package.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  "dependencies": {
+    "@turf/circle": "^6.5.0",
+    "@turf/transform-rotate": "^6.5.0",
+    "@zeit/next-css": "^1.0.1",
+    "csvtojson": "^2.0.10",
+    "intersection-observer": "^0.12.0",
+    "mapbox-gl": "^1.13.1",
+    "next": "9.5.4",
+    "next-images": "^1.7.0",
+    "react": "^16.14.0",
```
Comment (lacabra) [inline packages/map/package.json:23]: `react` and `react-dom` are already included in the parent `package.json` do they should not be duplicated here

**Review by lacabra — CHANGES_REQUESTED** [review]
The following files:
* packages/map/map/favicon.ico
* packages/map/map/github.jpeg
* packages/map/map/github.png
* packages/map/map/globe.png
* packages/map/map/hardware.svg

Are duplicates of their corresponding counterparts in `packages/map/public`. They should only exist in one place, and I believe `public` is where they belong. Thus, I would ask to remove these.

**Comment by Rolikasi** [comment]
> The following files (and all other assets in `packages/map/map`...)
> 
>     * packages/map/map/favicon.ico
> 
>     * packages/map/map/github.jpeg
> 
>     * packages/map/map/github.png
> 
>     * packages/map/map/globe.png
> 
>     * packages/map/map/hardware.svg
> 
> 
> Are duplicates of their corresponding counterparts in `packages/map/public`. They should only exist in one place, and I believe `public` is where they belong. Thus, I would ask to remove these.

Oh, that path supposed to be gitignored. It's renamed `/out`(like build/static) directory. I'll fix it tomorrow

File: packages/map/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+## 💻 Development Environment
+
+Setup your development environment as follows:
+
+1. Clone this repo:
+    - SSL:
+    ```bash
+    git clone git@github.com:rolikasi/digital-public-goods-map
```
Comment (lacabra) [inline packages/map/README.md:60]: Update this to `unicef/publicgoods-scripts`?

File: packages/automation/index.js [hunk 3]
```diff
@@ -456,5 +458,16 @@ textLine.each(function (d) {
     console.log('Modified files:', changedFiles.join(', '));
     fs.copyFileSync(pathFormHtml, destFormHtml);
   });
-
+  fs.readFile(pathMapHtml,'utf8',  function (err, html) {
```
Comment (Rolikasi) [inline packages/automation/index.js:461]: From this line script reads `publicgoods-website/map/index.html`. Then it splits and saves parts of html into multiple files. These files are used in `publicgoods-scripts/packages/map/pages/_document.js` to create a template page with wordpress styles.

File: packages/map/pages/_document.js [hunk 4]
```diff
@@ -1,296 +1,26 @@
...
-   _gaq.push(['_setAccount', 'UA-143534016-1']);
-   _gaq.push(['_trackPageview']);
-   (function() {
-   var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-   ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
-   var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
-   })();`,
-            }}
-          />
+          <html dangerouslySetInnerHTML={{__html: head}} />
```
Comment (Rolikasi) [inline packages/map/pages/_document.js:16]: This is how we place parts of html files from `publicgoods-website/map/index.html` before rendering nextJS app. [Reference](https://stackoverflow.com/a/60362264)

**Review by lacabra — APPROVED** [review]
Looks good to me. Thanks @Rolikasi for this amazing piece of work 💪 🚀

**Pull request merged**